File: app/src/main/java/com/app/activities/MainActivity.java
package com.app.activities;

import androidx.appcompat.app.AppCompatActivity;
import androidx.recyclerview.widget.DividerItemDecoration;
import androidx.recyclerview.widget.LinearLayoutManager;
import androidx.recyclerview.widget.RecyclerView;

import android.content.Intent;
import android.os.Bundle;
import android.widget.ImageView;

import com.app.R;
import com.app.api.models.Movie;
import com.app.api.models.Similar;
import com.app.mvp.main.Interfaces.IMainPresenter;
import com.app.mvp.main.Interfaces.IMainView;
import com.app.mvp.main.MainPresenter;
import com.app.utils.Constants;
import com.bumptech.glide.Glide;

import java.util.ArrayList;

public class MainActivity<LineAdapter> extends AppCompatActivity implements IMainView {

    RecyclerView mRecyclerView;
    private LineAdapter mAdapter;
    ImageView header_image;
    RecyclerView similar_list;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        new MainPresenter(this);

    }

    @Override
    public void setupHeader(Movie movie) {
        header_image = findViewById(R.id.header_image);
        similar_list = findViewById(R.id.similar_list);
        similar_list.setNestedScrollingEnabled(false);

        Glide.with(this)
                .load(Constants.url_img+movie.getPoster_path())
                .centerCrop()
                .into(header_image);
    }

    @Override
    public void setupRecycleSimilar(Similar similar) {
//        LinearLayoutManager layoutManager;
//        layoutManager = new LinearLayoutManager(this);
//        mRecyclerView.setLayoutManager(layoutManager);
//
//        mAdapter = new LineAdapter(new ArrayList<>(0));
//        mRecyclerView.setAdapter(mAdapter);
//
//        mRecyclerView.addItemDecoration(
//                new DividerItemDecoration(this, DividerItemDecoration.VERTICAL));
    }
}

File: app/src/main/java/com/app/activities/MainActivity.kt
package com.app.activities

import android.os.Bundle
import android.widget.ImageView
import android.widget.TextView
import androidx.appcompat.app.AppCompatActivity
import androidx.recyclerview.widget.LinearLayoutManager
import androidx.recyclerview.widget.RecyclerView
import com.app.R
import com.app.api.models.Movie
import com.app.api.models.Similar
import com.app.mvp.main.Interfaces.IMainView
import com.app.mvp.main.MainPresenter
import com.app.utils.Constants
import com.bumptech.glide.Glide

class MainActivity : AppCompatActivity(), IMainView {
    private var mAdapter: RecycleViewAdapter? = null
    var header_image: ImageView? = null
    var similar_list: RecyclerView? = null
    var txtTitle: TextView? = null
    var txtLikes: TextView? = null
    var txtPopularity: TextView? = null
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        MainPresenter(this)
    }

    override fun setupHeader(movie: Movie) {
        header_image = findViewById(R.id.header_image)
        similar_list = findViewById(R.id.similar_list)
        txtTitle = findViewById(R.id.txtTitle)
        txtPopularity = findViewById(R.id.txtPopularity)
        txtLikes = findViewById(R.id.txtLikes)

        similar_list?.isNestedScrollingEnabled = false
        txtTitle?.text  = movie.original_title
        txtPopularity?.text  = movie.popularity.toString() + "Watched"
        txtLikes?.text = movie.vote_count.toString() + " Likes"

        header_image?.let {
            Glide.with(this)
                .load(Constants.url_img + movie.poster_path)
                .centerCrop()
                .into(it)
        }
    }

    override fun setupRecycleSimilar(similar: Similar) {
        val layoutManager: LinearLayoutManager
        layoutManager = LinearLayoutManager(this)
        similar_list?.layoutManager = layoutManager
        mAdapter = RecycleViewAdapter(similar.results)
        similar_list?.adapter = mAdapter
    }
}